src/google/protobuf/stubs/io_win32.cc

```diff
@@ -355,6 +360,92 @@ wstring testonly_utf8_to_winpath(const char* path) {
   return as_windows_path(path, &wpath) ? wpath : wstring();
 }
 
+bool expand_wildcards(
+    const string& path, std::function<void(const string&)> consume) {
+  if (path.find_first_of("*?") == string::npos) {
+    // There are no wildcards in the path, we don't need to expand it.
+    consume(path);
+    return ExpandWildcardsResult::kSuccess;
+  }
+
+#ifdef SUPPORT_LONGPATHS
```

Comment:
Is there a way to unify some of this logic between the `SUPPORT_LONGPATHS` and non-`SUPPORT_LONGPATHS` cases? It seems like the logic is mostly the same, so I wonder if templates would make it possible to unify the two cases.